pkg/image/prune/imagepruner.go

```diff
@@ -488,10 +557,33 @@ func (p *imagePruner) Run(pruneImage ImagePruneFunc, pruneStream ImageStreamPrun
 	allNodes := p.g.NodeList()
 
 	imageNodes := imageNodeSubgraph(allNodes)
+	if len(imageNodes) == 0 {
```

Comment:
> I can't think of any reason to create synthetic nodes in the image pruning graph. We only care about images managed by our registry in this instance.

The question is more along the lines of wanting a pretty standard way to create edges through the graph.  Once you can do that and you have a very easy way to create all edges, then pruning is simply building out the entire graph using a normal graph builder mechanism. After the generic graph (with all links, including those to synthetic nodes) is created, then you create a subgraph based on the edge kinds you're interested in traversing.

I'm ok doing it in a later refactor if we need to, I just wasn't sure why looking at imageNodes is preferable to looking at options.Images.